x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/planner/QueryFolder.java

```diff
@@ -115,6 +119,26 @@ protected PhysicalPlan rule(LimitWithOffsetExec limit) {
         }
     }
 
+    private static class PropagateCompositeKeys extends FoldingRule<SampleExec> {
```

Comment:
This seems better suited as a logical optimization rule as a oppose to a folding one.

Reply:
Actually, this is a leftover from one of the initial ideas on how to introduce bucket extractors. The name of this rule is a misnomer, it doesn't actually propagate the  composite keys, but more creating extractors. I've removed this rule, the logic that's adding the extractors is in [ExecutionManager](https://github.com/elastic/elasticsearch/pull/85206/files#diff-c0a6ebe60abcb6c311669f3bddbe5642dbbc847c4aad7de13ff590ac72e04ed5R161).